component/conprof/http/api.go

```diff
@@ -209,11 +216,18 @@ func queryGroupProfiles(c *gin.Context) ([]GroupProfiles, error) {
 			case topology.ComponentTiFlash:
 				compNum.TiFlash = num
 			}
+			totalCompNum += num
 		}
+
+		state := ProfilingStateFinished
+		if ts == lastTS && totalCompNum < len(components) {
```

Comment:
I don't think this logic is correct. It may be possible that the scrape is finished but the component list has just been changed (e.g. a new component is added). Then you incorrectly displayed a finished job as running.

Reply:
Nice catch, already resolved this bug now.
The `components` is get from `GetRunningScrapeComponents`, and the function only returns the components which already start to scraping.